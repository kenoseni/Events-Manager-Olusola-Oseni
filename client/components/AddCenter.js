import React, { Component } from 'react';
import PropTypes from 'prop-types';
import SubmitButton from './SubmitButton';
import CenterImage from './CenterImage';
import * as centerActions from '../actions/CenterActions';
import { eventCenters } from '../reducers';
import { validateCenterInput } from './Validations';

class AddCenter extends Component {
  constructor(props) {
    super(props);
    this.state = {
      center: {},
      image: '',
      error: {}
    };
    this.submit = this.submit.bind(this);
    this.getInput = this.getInput.bind(this);
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.image.image !== this.props.image.image) {
      this.setState({ image: nextProps.image.image });
    }
  }

  isValid() {
    const { error, isValid } = validateCenterInput(this.state);
    if (!isValid) {
      this.setState({ error });
    }
    return isValid;
  }

  getInput(e) {
    this.setState({
      center: {
        ...this.state.center,
        [e.target.name]: e.target.value
      }
    });
  }

  submit(e) {
    e.preventDefault();
    this.setState({ error: {} });
    const {
      centerName,
      centerDescription,
      centerLocation,
      centerPrice,
      centerFacilities,
      centerCapacity,
      error
    } = this.state.center;
    const { image } = this.state;
    const { history } = this.props;
    if (this.isValid()) {
      this.props
        .addCenter({
          name: centerName,
          description: centerDescription,
          location: centerLocation,
          price: centerPrice,
          facilities: centerFacilities,
          capacity: centerCapacity,
          image
        })
        .then(() => {
          if (this.props.eventCenters.error.message !== undefined) {
            this.setState({ error: this.props.eventCenters.error });
          } else {
            $('#addCenter').modal('hide'), history.push('/centers');
          }
        });
    }
  }

  render() {
    const { error } = this.state;
    const { addImage } = this.props;
    return (
      <div>
        <div className="container">
          {/*<!-- Add Center Modal -->*/}
          <div
            className="modal fade right"
            id="addCenter"
            role="dialog"
            aria-labelledby="addCenter"
          >
            <div className="modal-dialog modal-full-height modal-right">
              {/*<!-- Modal content-->*/}
              <div className="modal-content">
                <div id="addCenterModal" className="modal-header">
                  <h4
                    id="addCenterTitle"
                    className="modal-title text-center font-weight-bold"
                  >
                    {this.props.title}
                  </h4>
                  <button type="button" className="close" data-dismiss="modal">
                    &times;
                  </button>
                </div>
                <div className="modal-body">
                  <form
                    action=""
                    method=""
                    role="form"
                    id="addForm"
                    onSubmit={this.submit}
                  >
                    <div id="imageFile" className="">
                      <div className="input-group mb-3">
                        <CenterImage addImage={addImage} />
                      </div>
                      {error.message && (
                        <div className="alert alert-danger">
                          {error.message}
                        </div>
                      )}
                      <div className="">
                        <div className="container-fluid">
                          <div className="row">
                            <div className="col-12">
                              <div id="addCenterForm" className="">
                                <span className="" id="sizing-addon1">
                                  Name:
                                </span>
                                <input
                                  type="text"
                                  name="centerName"
                                  id="addCenterName"
                                  onChange={this.getInput}
                                  className="form-control"
                                  aria-describedby="sizing-addon1"
                                />
                              </div>
                            </div>
                          </div>
                          <div className="row">
                            <div className="col-12">
                              <div className="">
                                <span
                                  id="centerDescription"
                                  className=""
                                  id="sizing-addon1"
                                >
                                  Description:
                                </span>
                                <input
                                  type="text"
                                  id="addCenterDescription"
                                  name="centerDescription"
                                  onChange={this.getInput}
                                  className="form-control"
                                  aria-describedby="sizing-addon1"
                                />
                              </div>
                            </div>
                          </div>
                          <div className="row">
                            <div className="col-12">
                              <div className="">
                                <span className="" id="sizing-addon1">
                                  Location:
                                </span>
                                <input
                                  type="text"
                                  id="addCenterLocation"
                                  name="centerLocation"
                                  onChange={this.getInput}
                                  className="form-control"
                                  aria-describedby="sizing-addon1"
                                />
                              </div>
                            </div>
                          </div>
                          <div className="row">
                            <div className="col-12">
                              <div className="">
                                <span className="" id="sizing-addon1">
                                  Price:
                                </span>
                                <input
                                  type="text"
                                  id="addCenterPrice"
                                  name="centerPrice"
                                  onChange={this.getInput}
                                  className="form-control"
                                  aria-describedby="sizing-addon1"
                                />
                              </div>
                            </div>
                          </div>
                          <div className="row">
                            <div className="col-12">
                              <div className="">
                                <span className="" id="sizing-addon1">
                                  Facilities:
                                </span>
                                <input
                                  type="text"
                                  id="addCenterFacilities"
                                  name="centerFacilities"
                                  onChange={this.getInput}
                                  className="form-control"
                                  aria-describedby="sizing-addon1"
                                />
                              </div>
                            </div>
                          </div>
                          <div className="row">
                            <div className="col-12">
                              <div className="">
                                <span className="" id="sizing-addon1">
                                  Capacity:
                                </span>
                                <input
                                  type="text"
                                  id="addCenterCapacity"
                                  name="centerCapacity"
                                  onChange={this.getInput}
                                  className="form-control"
                                  placeholder=""
                                  aria-describedby="sizing-addon1"
                                />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <br />
                    <SubmitButton id="submitButton" name="ADD CENTER" />
                  </form>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
AddCenter.propTypes = {
  title: PropTypes.string.isRequired,
  addCenter: PropTypes.func.isRequired
};

export default AddCenter;
