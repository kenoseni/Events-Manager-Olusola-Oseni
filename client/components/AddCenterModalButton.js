import React, {Component} from 'react';

class AddCenterModalButton extends Component {

  render() {
    return (
      <div>
        <button className="float" data-toggle="modal" data-target="#addCenter">
          <i className="fa fa-plus my-float"></i>
        </button>
      </div>
    ) 
  }
}
export default AddCenterModalButton;