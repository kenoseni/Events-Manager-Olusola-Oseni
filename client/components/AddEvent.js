import React, {Component} from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import SubmitButton from './SubmitButton';

class AddEvent extends Component {
  constructor(props) {
    super(props)
    this.state = {
      value: 'Choose a Center',
      event: {},
      centers: []
    }
    this.submit = this.submit.bind(this)
    this.getInput = this.getInput.bind(this)
  }
  componentWillReceiveProps(nextProps) {
    if (nextProps.eventCenters.centers !== this.props.eventCenters.centers) {
      this.setState({centers: nextProps.eventCenters.centers})
    }
  }
  
  getInput(e) {
    this.setState({ event: {
      ...this.state.event,
      [e.target.name]: e.target.value
    },
   })
  }
  
  submit(e) {
    e.preventDefault();
    const {name, centerId, date, time} = this.state.event
    $('#addEvent').modal('hide')
    this.props.addEvent({
      name,  
      centerId, 
      date, 
      time
    })
  }
  render() {
    const {centers} = this.state
    return (
      <div>
        <div className="container">            
          {/*<!-- Add Events Modal -->*/}
          <div className="modal fade" id="addEvent" role="dialog">
            <div className="modal-dialog modal-lg">      
              {/*<!-- Modal content-->*/}
              <div className="modal-content">
                <div className="modal-header">
                  <h4 className="modal-title text-center font-weight-bold">{this.props.title}</h4>
                  <button type="button" className="close" data-dismiss="modal">&times;</button>   
                </div>
                <div className="modal-body">
                  <form action="" method="" role="form" onSubmit={this.submit}>       
                    <div className="card">
                      <div className="card-body text-center">
                        <div className="input-group input-group-sm">
                          <span className="input-group-addon" id="sizing-addon1">Event Name</span>
                          <input type="text" name="name" onChange={this.getInput} className="form-control" aria-describedby="sizing-addon1" />
                        </div>
                      </div>
                      <div className="card-body">
                        <div className="container-fluid">
                          <div className="row">
                            <div className="col-xs-12">
                              <div className="input-group input-group-sm">
                                <span className="input-group-addon" id="sizing-addon1">Location</span>
                                <select className="form-control" name="centerId" onChange={this.getInput}>
                                  <option value={this.state.value}>Choose a Center</option>
                                  {
                                    centers.map(center => <option key={center.id} value={center.id}> {(center) ? center.name.toUpperCase(): ''} </option>)
                                  }
                                </select>
                              </div>
                            </div>
                            <div className="col-xs-12">
                              <div className="input-group input-group-sm">
                                <span className="input-group-addon" id="sizing-addon1">Date</span>
                                <input type="date" name="date" onChange={this.getInput} className="form-control" aria-describedby="sizing-addon1" />
                              </div>
                            </div>
                            <div className="col-xs-12">
                              <div className="input-group input-group-sm">
                                <span className="input-group-addon" id="sizing-addon1">Time</span>
                                <input type="time" name="time" onChange={this.getInput} className="form-control" aria-describedby="sizing-addon1" />
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                    </div>
                    <br />
                    <SubmitButton name='ADD EVENT'/>
                  </form>            
                </div>
              </div> 
            </div>
          </div>
        </div>
      </div>
    )
  }
}
export default AddEvent;
