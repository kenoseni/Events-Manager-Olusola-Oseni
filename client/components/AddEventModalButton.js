import React, {Component} from 'react';

class AddEventModalButton extends Component {

  render() {
    return (
      <div>
        <button className="float" data-toggle="modal" data-target="#addEvent">
          <i className="fa fa-plus my-float"></i>
        </button>
      </div>
    ) 
  }
}
export default AddEventModalButton;
