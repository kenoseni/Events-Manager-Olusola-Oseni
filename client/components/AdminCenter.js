import React, {Component} from 'react';
import {Link} from 'react-router-dom';
import DeleteModal from './DeleteModal';
import jwtDecode from 'jwt-decode';
import ModifyCenter from './ModifyCenter'


class AdminCenter extends Component {
  constructor(props) {
    super(props); 
  }

  componentWillMount() {
    let token = localStorage.getItem('x-access-token');
    if (token) {
      let decodedToken = jwtDecode(token);
      let isAdmin = decodedToken.isadmin;
      let role = decodedToken.userrole

      if(isAdmin) {
      this.isAdmin = true
      }
    }  
  }
  render () {
    const {center, i} = this.props
    return (
      <div className="col-md-4 mb-4">
        <div className="card w-40">
          <div className="view overlay hm-white-slight">
            <Link to={`/centers/${center.id}`}>
              <img src={center.image} className="img-fluid" alt={center.name} id="img-preview" style={{width:'100%', height:'250px'}} />
            </Link>
          </div>
          <div className="card-body">
            <h5 className="card-title font-weight-bold text-center">{center.name.toUpperCase()}</h5>
            <p className="card-text font-weight-bold ">Description: {center.description}</p>
            <p className="card-text font-weight-bold">Location: {center.location}</p>
            <p className="card-text font-weight-bold">Price: {center.price}</p>
            <p className="card-text font-weight-bold">Facilities: {center.facilities}</p>
            <p className="card-text font-weight-bold">Capacity: {center.capacity} guests</p>
            <p className="card-text font-weight-bold">
              Availability: 
              {(center.avaliability === true) ? 
                ' Center Available': 
                ' Currently not Available'
              } 
            </p>                                            
          </div>
          {this.isAdmin &&
            <div className="card-footer bg-transparent">
              <ul className="nav">
                <li>
                  <button data-toggle="modal" data-target={`#${center.id}`}>
                    <i className="fa fa-pencil-square-o fa-lg nav-link" aria-hidden="true"></i>
                  </button>
                </li>
                <li>
                  <button data-toggle="modal" data-target={`#${i}`}>
                    <i className="fa fa-trash-o float-right fa-lg nav-link" aria-hidden="true"></i>
                  </button>
                </li>
              </ul>
            </div>
          }
        </div>
        <ModifyCenter title='Modify Center' i={i} center={center} {...this.props} />
        <DeleteModal 
          title={`Are you sure you want to delete ${center.name} center?`} 
          i={i} 
          center={center} 
          {...this.props} 
        /> 
      </div>
    )
  }
}

export default AdminCenter;
