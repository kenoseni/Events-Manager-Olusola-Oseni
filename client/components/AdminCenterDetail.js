import React, {Component} from 'react';
import {link} from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux'
import * as centerActions  from '.././actions/CenterActions'
import { eventCenters } from '.././reducers/';
import Navbar2 from './Navbar2';
import AdminCenterEvents from './AdminCenterEvents'

class AdminCenterDetail extends Component {
  constructor(props) {
    super(props);
  }

  render () {
    const { center } = this.props;
    return (
      <div>
        <Navbar2 />
        <main>
          <div className="container">
            <div className="row" id="info">
              <div className="col-xs-12 col-md-4">
                <div className="card w-40">  
                  {/*<!--Card image-->*/}
                  <div className="view overlay hm-white-slight">
                    <img src="./img2.jpg" className="img-fluid" alt={center.name} style={{width: '100%', height: '250px'}} />
                  </div>   
                  {/*<!--Card content-->*/}
                  <div className="card-body">
                    {/*<!--Title-->*/}
                    <h5 className="card-title font-weight-bold text-center">{(center.name === undefined ? null: center.name.toUpperCase())}</h5>
                    {/*<!--Text-->*/}
                    <p className="card-text"></p>
                  </div>     
                </div>
                {/*<!--/.Card-->*/}   
              </div>
              <div className="col-xs-12 col-md-8" id="paradise">
                <div className="card" >
                  <div className="card-header text-right">
                    Details 
                  </div>
                  <div className="card-body">
                    <h4 className="card-title">Description</h4>
                    <h5 className="card-text">{center.description}</h5>
                    <hr />
                    <h4 className="card-text">Facilities</h4>
                    <ul>
                      <li>{center.facilities}</li>
                    </ul>
                  </div>
                </div>
              </div>
            </div>
            <div className="row" id="space">
              <div className="col-xs-12 col-md-4" id="space2">
                <div className="card w-40">  
                  {/*<!--Card image-->*/}
                  <div className="view overlay hm-white-slight">
                    <img src="./img1.jpg" className="img-fluid" alt={center.name} style={{width: '100%', height: '250px'}} />
                  </div>   
                </div>
                {/*<!--/.Card-->*/}
              </div>
              <AdminCenterEvents/>
            </div>
          </div>
        </main>
      </div>
    )
  }
}

export default AdminCenterDetail;
