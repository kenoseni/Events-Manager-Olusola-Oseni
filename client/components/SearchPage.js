import React, {Component} from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import * as centerActions  from '../actions/CenterActions'
import {Link} from 'react-router-dom';
import AdminCenter from './AdminCenter';
import NavBar from './Navbar';
import Pages from './Pages';

export class SearchPage extends Component {
  constructor(props) {
    super(props); 
  }
  // componentDidMount() {
  //   const page = this.props.history.location.search.split('=')[1];
  //   this.props.getCenters(page);
  // }
  render () {
    const { searchedCenters, count, limit} = this.props.eventCenters;
    const { history } = this.props
    
    return (
      <div style={{paddingTop: '90px'}}>
        <NavBar home='Home' centers='Centers' events='Events'
          login='Log In' signup='Sign Up' admin='Admin' page='Home' logout='Log Out'
        />
        <h3 style={{padding: '30px 0px 0px 150px'}}>Search Results for Centers:</h3>
        <div className="container">
          <div className="row" id="centers">
            {(searchedCenters.length > 0) ?
              searchedCenters.map((center, i) => <AdminCenter {...this.props} key={center.id} i={i * -1} center={center} />) : 
              <h1 style={{marginTop:'200px', paddingLeft: '300px'}}>Oh Snap, No Result for your Search</h1>
            }
          </div>
        </div>
        {count !== undefined && <Pages count={count} history={history} limit={limit} />}
      </div>
    )
  }
}

const mapStateToProps = (state) => ({
  eventCenters: state.eventCenters,
});

const mapDispatchToProps = (dispatch) => bindActionCreators({
  ...centerActions, 
}, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(SearchPage);
