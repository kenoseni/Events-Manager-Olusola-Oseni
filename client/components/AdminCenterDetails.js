import React, {Component} from 'react';
import {link} from 'react-router-dom';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux'
import PropTypes from 'prop-types';
import * as centerActions from '.././actions/CenterActions'
import { eventCenters } from '.././reducers/';
import AdminCenterDetail from './AdminCenterDetail';

export class AdminCenterDetails extends Component {
  constructor(props) {
    super(props);
    this.state = {
      center: this.props.eventCenters.center
    }
  }
  componentWillMount() {
    let id = this.props.match.params.id;
    this.props.getOneCenter(id);
  }
  
  componentWillReceiveProps(nextProps){
    if (nextProps.eventCenters.center !== this.props.eventCenters.center) {
      this.setState({
        center: nextProps.eventCenters.center
      })
    }
  }
  render () {
    const { center } = this.state;
    const { getOneCenter, searchForCenters} = this.props;
    const { id } = this.props.match.params;
    
    return (
      <div>
        {center !== {} && 
          <AdminCenterDetail 
            center={center} 
            getOneCenter={getOneCenter} 
            id={id} searchForCenters={searchForCenters} 
          />
        }
      </div>
    )
  }
}
AdminCenterDetails.propTypes = {
  getOneCenter: PropTypes.func.isRequired,
}
const mapStateToProps = state => ({
  eventCenters: state.eventCenters
});

const mapDispatchToProps = dispatch => bindActionCreators(centerActions, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(AdminCenterDetails)
