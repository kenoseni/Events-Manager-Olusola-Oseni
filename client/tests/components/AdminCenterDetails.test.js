import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {SearchPage} from '../../components/SearchPage';
import { AdminCenterDetails} from '../../components/AdminCenterDetails';

configure({ adapter: new Adapter() });
const mockgetOneCenter = jest.fn()
const mocksearchForCenters = jest.fn()

let props = {

  eventCenters: {
      center: {}
  },
  match: {
      params: '1'
  }
}

const wrapper = shallow(<AdminCenterDetails
  getOneCenter={mockgetOneCenter}
  searchForCenters={mocksearchForCenters}
  {...props}
/>)

describe('<AdminCenterDetails />', () => {
  it('should render a <AdminCenterDetails /> component', () => {
    expect(wrapper).toMatchSnapshot();
  });
})
