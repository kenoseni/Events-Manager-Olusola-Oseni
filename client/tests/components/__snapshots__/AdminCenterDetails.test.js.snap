// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdminCenterDetails /> should render a <AdminCenterDetails /> component 1`] = `
<div>
  <AdminCenterDetail
    center={Object {}}
    getOneCenter={
      [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
      }
    }
    searchForCenters={[MockFunction]}
  />
</div>
`;
